Update dataset

sihu opened the pull request.

sihu committed:
Update dataset

sihu commented on data/PLZO_CSV_WGS84.csv:
```diff
@@ -1,4139 +1,5664 @@
+Wabern;3084;0;Köniz;355;BE;7.449534817204611;46.928587500902175;de
+Spiegel b. Bern;3095;0;Köniz;355;BE;7.43293430753073;46.92556695161132;de
+Liebefeld;3097;0;Köniz;355;BE;7.415781571292609;46.929726140639936;de
+Köniz;3098;0;Köniz;355;BE;7.435272380867865;46.93035927807075;de
```
hmm...i just copies the file but it seems that something with the encoding is weird

rnestler committed:
Remove BOM

BOM doesn't make sense in a UTF-8 document.

coorasse reviewed (approved):
I don't know much about this project, I guess this PR is part of...something? Fine for me.

rnestler committed:
Update dataset to latest version

rnestler committed:
Fix encoding when reading the data file

rnestler committed:
Update specs

Aadorf exists now in two different cantons, but with the same PLZ for
some reason:

    Aadorf;8355;0;Elgg;294;ZH;8.89224905677294;47.49195782198037;de
    Aadorf;8355;0;Aadorf;4551;TG;8.89769823294554;47.49135943297217;de

rnestler committed:
Adapt longitude latitude specs

The updated data seems to improve precision, so that they are unique.

rnestler commented on lib/swiss_village_directory/repository.rb:
```diff
@@ -7,7 +7,7 @@ class Repository
     def initialize
       filename = "#{File.dirname(__FILE__)}/../../data/PLZO_CSV_WGS84.csv"
-      options = { col_sep: ';', encoding: 'ISO-8859-1:UTF-8', headers: :first_row }
+      options = { col_sep: ';', encoding: 'UTF-8', headers: :first_row }
```
The encoding used to be `ISO-8859-1` for words with umlauts. This changed to be just `UTF-8` which is cleaner anyways.

rnestler commented on spec/swiss_village_directory/repository_spec.rb:
```diff
@@ -39,9 +39,9 @@
+      expect(repo.find_all_by(longitude: 9.595055059743).count).to eq(1)
       expect(repo.find_all_by(latitude: 1000).count).to eq(0)
-      expect(repo.find_all_by(latitude: 47.491).count).to eq(6)
+      expect(repo.find_all_by(latitude: 45.86277462745).count).to eq(1)
```
Searching by latitude and expecting multiple results won't work anymore, since the precision got increased to be unique.

sihu replied:
that would be valuable as a comment in the spec i think

sihu reviewed (commented):
LGTM 👍
